devel/libssh2/Portfile

```diff
@@ -1,42 +1,43 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
 PortSystem          1.0
-PortGroup           muniversal 1.0
+PortGroup           github      1.0
+PortGroup           muniversal  1.0
+PortGroup           openssl     1.0
+
+github.setup        libssh2 libssh2 1.10.0 libssh2-
+github.tarball_from releases
+revision            0
 
-name                libssh2
-version             1.9.0
-revision            1
 categories          devel net
 platforms           darwin
 maintainers         {wohner.eu:normen @Gminfly} openmaintainer
 
+homepage            https://www.libssh2.org/
+
 description         libssh2 is a library implementing the SSH2 protocol
+
 long_description    libssh2 is a library implementing the SSH2 protocol as \
                     defined by Internet Drafts: SECSH-TRANS(22), \
                     SECSH-USERAUTH(25), SECSH-CONNECTION(23), SECSH-ARCH(20), \
                     SECSH-FILEXFER(06), SECSH-DHGEX(04), and SECSH-NUMBERS(10).
 
 license             BSD
 
-homepage            https://www.libssh2.org/
-master_sites        ${homepage}download/
-
-checksums           rmd160  eb3553a9b2c05d5b6a24159db8a1478f9aea3877 \
-                    sha256  d5fb8bd563305fd1074dda90bd053fb2d29fc4bce048d182f96eaa466dfadafd \
-                    size    888551
+checksums           rmd160  b3af89cc5974dbffbcba0dd80dcd521e6e60b39d \
+                    sha256  2d64e90f3ded394b91d3a2e774ca203a4179f69aebee03003e5a6fa621e41d51 \
+                    size    965044
 
 if {${os.platform} eq "darwin" && ${os.major} < 10} {
     # remove errant pragmas inside functions no supported on older gcc versions
     patchfiles-append patch-libssh2-pragmas-older-gcc.diff
 }
 
-depends_lib         path:lib/libssl.dylib:openssl port:zlib
+depends_lib         port:zlib
```

Comment:
this should likely become `depends_lib-append` as not to overwrite dependencies from the PG (e.g. `openssl`)?

Reply:
ah, well that's good! There might be some magic happening that it does that somehow after "finishing" the rest of the Portfile. Otherwise, I would think that that dependency is added when you include the PG and then the use of `depends_lib` would overwrite it... In any case, using `depends_lib-append` never hurts ;)